source/getting-started/core-concepts.md

```diff
@@ -1,58 +1,136 @@
+templates. Anything that is valid Handlebars syntax is valid Ember syntax.
+
+Each template has a context, and properties you access are read off that context.
+For example this template shows a welcome message with a name:
+
+```handlebars
+<div>Hi {{name}}, this is a valid Ember template!</div>
+```
+
+The mustache of this template (the `{{}}`) refers to some dynamic content.
+Besides properties, you can also use helpers and components inside a mustache.
+
+## Helpers
+
+Ember Helpers are another way to generate a value in templates. A helper
+is a peice of logic that returns a single value (much like a property
+has only a single value).
+
+For example, this helper and accompianing usage create a full name:
+
+```js
+// app/helpers/full-name.js
+import Ember from 'ember';
+
+export default Ember.Helper.helper(function([firstName, lastName]) {
+  return `Mrs. ${firstName} ${secondName}`;
+});
+```
+
+```handlebars
+{{full-name firstName "Doe"}}
+```
+
+You can read more about helpers in [this API documentation](http://emberjs.com/api/classes/Ember.Helper.html).
 
 ## Components
 
 Components are the primary way user interfaces are organized in Ember. They
-consist of two parts: a template, and a source file written in JavaScript that
-defines the component's behavior. For example, a blog application might have a
-component for displaying a list of blog posts called `all-posts`, and another
-component for displaying an individual post called `view-post`. If users can
-upvote a post, the `view-post` component might define a behavior like _when the
-user clicks the upvote button, increase the `vote` property's value by 1_.
+consist of two parts:
 
-## Controllers
+* A template, for example `app/templates/components/welcome-message.hbs`
+* Optionally a source file written in JavaScript, for example
+  `app/components/welcome-message.js`.
+* **Components must contain a - in their name**. More about that later.
+
+Components differ from helpers in that they can output not only a value, but
+also manage DOM. An important property of components is that they are **isolated**.
+For example, a component's DOM is called "shadow" DOM.
+
+In this example, a component's template adds a div around some other DOM:
 
-Controllers are very much like components, so much so that in future versions of
-Ember, controllers will be replaced entirely with components. At the moment,
-components cannot be routed to (see below), but when this changes, it will be
-recommended to replace all controllers with components.
+```handlebars
+{{! app/templates/components/my-component.hbs }}
+<div class="inner">
+  {{yield}}
+</div>
+```
+
+```handlebars
+<span>Hello</span>
+<div class="outer">
+  {{#my-component}}
+    Hello there!
+  {{/my-component}}
+</div>
+```
+
+Because components deal with DOM, they are also responsible for maintaining
+the bulk of an application's state. Additionally they handle events like
+clicks, keyUp, and other more semtantic actions.
+
+## The Router
```

Comment:
I've seen this over and over and over again when I've taught relatively beginner programmers their first framework (usually Rails or Ember). When you leave out a step in explaining how things actually work because conventions hide some of the complexity some of the time, the short-term gain of one fewer step causes much more pain when people have a broken mental model. For you or me, learning "URL -> router -> template" and then being told, "Hey, just kidding, it's actually URL -> router -> route -> template/model" is fine, but for people with relatively little experience, it's very difficult for them to change their mental model once it's built. That leads to all sorts of problems, like somebody loading a model in one route, and then expecting it to be available to a template in another route, when no model is loaded from the second route. I'm not making this stuff up - I've seen it firsthand.